engine/src/main/java/org/pentaho/di/job/entries/sftp/SFTPClient.java

```diff
@@ -0,0 +1,479 @@
+    }
+  }
+
+  public void chdir( String dirToChangeTo ) throws KettleJobException {
+    try {
+      c.cd( dirToChangeTo.replace( "\\\\", "/" ).
+        replace( "\\", "/" ) );
+    } catch ( SftpException e ) {
+      throw new KettleJobException( e );
+    }
+  }
+
+  public String[] dir() throws KettleJobException {
+    String[] fileList = null;
+
+    try {
+      java.util.Vector<?> v = c.ls( "." );
+      java.util.Vector<String> o = new java.util.Vector<String>();
+      if ( v != null ) {
+        for ( int i = 0; i < v.size(); i++ ) {
+          Object obj = v.elementAt( i );
+          if ( obj != null && obj instanceof com.jcraft.jsch.ChannelSftp.LsEntry ) {
+            LsEntry lse = (com.jcraft.jsch.ChannelSftp.LsEntry) obj;
+            if ( !lse.getAttrs().isDir() ) {
+              o.add( lse.getFilename() );
+            }
+          }
+        }
+      }
+      if ( o.size() > 0 ) {
+        fileList = new String[o.size()];
+        o.copyInto( fileList );
+      }
+    } catch ( SftpException e ) {
+      throw new KettleJobException( e );
+    }
+
+    return fileList;
+  }
+
+  public void get( FileObject localFile, String remoteFile ) throws KettleJobException {
+    OutputStream localStream = null;
+    try {
+      localStream = KettleVFS.getOutputStream( localFile, false );
+      c.get( remoteFile, localStream );
+    } catch ( SftpException e ) {
+      throw new KettleJobException( e );
+    } catch ( IOException e ) {
+      throw new KettleJobException( e );
+    } finally {
+      if ( localStream != null ) {
+        try {
+          localStream.close();
+        } catch ( IOException ignore ) {
+          // Ignore any IOException, as we're trying to close the stream anyways
+        }
+      }
+    }
+  }
+
+  /**
+   * @deprecated use {@link #get(FileObject, String)}
+   * @param localFilePath
+   * @param remoteFile
+   * @throws KettleJobException
+   */
+  @Deprecated
+  public void get( String localFilePath, String remoteFile ) throws KettleJobException {
+    int mode = ChannelSftp.OVERWRITE;
+    try {
+      c.get( remoteFile, localFilePath, null, mode );
+    } catch ( SftpException e ) {
+      throw new KettleJobException( e );
+    }
+  }
+
+  public String pwd() throws KettleJobException {
+    try {
+      return c.pwd();
+    } catch ( SftpException e ) {
+      throw new KettleJobException( e );
+    }
+  }
+
+  public void put( FileObject fileObject, String remoteFile ) throws KettleJobException {
+    int mode = ChannelSftp.OVERWRITE;
+    InputStream inputStream = null;
+    try {
+      inputStream = KettleVFS.getInputStream( fileObject );
+      c.put( inputStream, remoteFile, null, mode );
+    } catch ( Exception e ) {
+      throw new KettleJobException( e );
+    } finally {
+      if ( inputStream != null ) {
+        try {
+          inputStream.close();
+        } catch ( IOException e ) {
+          throw new KettleJobException( e );
+        }
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 281:287, 299:305.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.